File: C_Programming/Competitive_Programming/Assignment4/program2.c
// Write a program which accept number from user and display its factor in Decresing order

// Input :  12
// Output : 6 4 3 2 1

#include <stdio.h>

void FactRev(int iNo)
{
    int iCnt = 0;
    int iAns = 1;
    int icnt = 0;

    for(iCnt = 1; iCnt < iNo; iCnt++)
    {
        if((iNo % iCnt) == 0)
        {   
           for(icnt = iCnt; icnt > 0; icnt--)
           {
                printf("%d\t",iCnt);
                break;
           } 
        }
    }
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    FactRev(iValue);

    return 0;
}

// Time Complexity : O(N)

File: Java_Programming/Competitive_Programming/Assignment20/program2.java
// Write a program to print number from N down to 1 in reverse order
class logic 
{
    void printReverse(int iNo)
    {
        int iCnt = 0;
        for(iCnt = iNo; iCnt >=1; iCnt--)
        {
            System.out.println(""+iCnt);
        }

    }
}

class program2
{
    public static void main(String A[] ) 
    {
        logic obj = new logic();
        obj.printReverse(10);
    }
}
